core/indexer/elasticsearch_test.go

```diff
@@ -230,51 +225,67 @@ func TestElasticIndexer_SaveMetaBlockNilHeader(t *testing.T) {
 		_ = logger.SetLogLevel("core/indexer:INFO")
 	}()
 
-	ei.SaveMetaBlock(nil, []uint64{0})
-	require.True(t, strings.Contains(output.String(), indexer.ErrNoHeader.Error()))
+	ei.SaveRoundInfo(indexer.RoundInfo{})
+	require.Empty(t, output.String())
 }
 
-func TestElasticIndexer_SaveRoundInfo(t *testing.T) {
+func TestElasticIndexer_SaveValidatorsPubKeys(t *testing.T) {
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
 
 	output := &bytes.Buffer{}
 	_ = logger.SetLogLevel("core/indexer:TRACE")
 	_ = logger.AddLogObserver(output, &logger.PlainFormatter{})
 	arguments := NewElasticIndexerArguments()
-	arguments.Marshalizer = &mock.MarshalizerMock{Fail: true}
 	arguments.Url = ts.URL
+	arguments.Marshalizer = &mock.MarshalizerMock{Fail: true}
 	ei, _ := indexer.NewElasticIndexer(arguments)
 
+	valPubKey := make(map[uint32][][]byte)
+
+	keys := [][]byte{[]byte("key")}
+	valPubKey[0] = keys
+	epoch := uint32(0)
+	ei.SaveValidatorsPubKeys(valPubKey, epoch)
+
 	defer func() {
 		_ = logger.RemoveLogObserver(output)
 		_ = logger.SetLogLevel("core/indexer:INFO")
 	}()
 
-	ei.SaveRoundInfo(indexer.RoundInfo{})
 	require.Empty(t, output.String())
 }
 
-func TestElasticIndexer_SaveValidatorsPubKeys(t *testing.T) {
+func TestElasticIndexer_EpochChange(t *testing.T) {
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
 
+	getEligibleValidatorsCalled := false
+
 	output := &bytes.Buffer{}
 	_ = logger.SetLogLevel("core/indexer:TRACE")
 	_ = logger.AddLogObserver(output, &logger.PlainFormatter{})
 	arguments := NewElasticIndexerArguments()
 	arguments.Url = ts.URL
 	arguments.Marshalizer = &mock.MarshalizerMock{Fail: true}
-	ei, _ := indexer.NewElasticIndexer(arguments)
-
-	valPubKey := make(map[uint32][][]byte)
+	arguments.ShardId = core.MetachainShardId
+	epochChangeNotifier := &mock.EpochStartNotifierStub{}
+	arguments.EpochStartNotifier = epochChangeNotifier
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	arguments.NodesCoordinator = &mock.NodesCoordinatorMock{
+		GetAllEligibleValidatorsPublicKeysCalled: func() (m map[uint32][][]byte, err error) {
```

Comment:
the epoch is not actually used. Could you please add the epoch to the header and also to the GetAllEligibleValidatorsPublicKeysCalled in order to validate that the NotifyAll was called with the correct epoch.

Reply:
done.

Reply:
ok